test/boost/sstable_compaction_test.cc

```diff
@@ -2114,6 +2114,29 @@ SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test) {
     });
 }
 
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test_valid_sstable) {
+    scrub_test test;
+
+    auto schema = test.schema();
+
+    auto muts = tests::generate_random_mutations(test.random_schema()).get();
+
+    test.run(schema, muts, [] (table_for_tests& table, compaction::table_state& ts, std::vector<sstables::shared_sstable> sstables) {
+        BOOST_REQUIRE(sstables.size() == 1);
+        auto sst = sstables.front();
+
+        // No way to really test validation besides observing the log messages.
+        sstables::compaction_type_options::scrub opts = {
+            .operation_mode = sstables::compaction_type_options::scrub::mode::validate,
+        };
+        table->get_compaction_manager().perform_sstable_scrub(ts, opts).get();
+
+        BOOST_REQUIRE(!sst->is_quarantined());
+        BOOST_REQUIRE_EQUAL(in_strategy_sstables(ts).size(), 1);
+        BOOST_REQUIRE_EQUAL(in_strategy_sstables(ts).front(), sst);
+    });
+}
+
```

Comment:
Will this test fail without the fixes?

Would be better to have a direct example of the failures rather than random generation.

Reply:
Yes, this test fails 2 out of 3 times without the fixes (either of them).
I prefer a random test like this, because it can reveal problems I didn't think of before.

Reply:
I agree with Avi that I prefer deterministic tests which try to enumerate the actual edge cases instead of hoping you run into these edge cases randomly, but I'm not against randomized tests if developers like them and it makes them write more tests :-)

Reply:
Another benefit of deterministic tests is that you can create them so each test reproduces the need for a *different* fix. In this PR, I counted four (!) different fixes to the validator, and what this test proves is that without any of them this test always fails, and with all of them it always passes. But how do you know that all four fixes are needed? It's better to have four different tests, each reproduces a different problem (e.g., one has a range tombstone, another has a promoted index, etc.).
But again, all that being said, I don't insist. This patch, and this test, is clearly a step in the right direction - even if it's not exactly the "perfect" methodology I might have recommended, it's definitely better than what we had until now - tests which were deterministic but didn't actually cover the important corner cases.